### ISSUE TITLE
Vulkan Layer_khronos_validation deteriorates rendering

### ISSUE THREAD

**Issue by Graou74** [LWJGL/lwjgl3#519]
### Environment

* LWJGL version: `2.3.2` or `2.3.3`
* LWJGL build #: `All I suppose (Gradle links)`
* Java version: `openjdk version "11.0.1" 2018-10-16`
* Platform: `macOS`
* Module: `Vulkan`
* Vulkan SDK: `vulkansdk-macos-1.1.126.0`

### Description

I am trying Vulkan with JavaFx, GLFW and the source code TriangleDemo.java.
I activate the layers by setting the '**validation**' variable to true.
`private static final boolean validation = true;`
But I make inaccessible the library '**libVkLayer_khronos_validation.dylib**' by renaming it '**MASK_libVkLayer_khronos_validation.dylib**'.
Then I launch the application and I get a GLFW window with the rendering of a triangle.
I also have an error message which tells me that the library '**libVkLayer_khronos_validation.dylib**' is missing.
```
testVulkan2.VulkanTest > test() STANDARD_ERROR
    ERROR OCCURED: dlopen(/Users/ ... /vulkansdk-macos-1.1.126.0/macOS/etc/vulkan/explicit_layer.d/../../../lib/libVkLayer_khronos_validation.dylib, 5): image not found
```
I do not have the use of the layer '**Layer_khronos_validation**' but the rendering corresponds exactly to what is expected.

Now I rename the library with its original name so that it is recognized during execution and I restart the program as it is.
I no longer have a console error, but there is no longer a triangle in the GLFW window. The background color is correct but the triangle has disappeared.

The same project on Windows works in all cases.

I put the project in attachment. It is an Eclipse project but it can be launched from the Gradle command line:
`./gradlew build -i`

You just have to change the two lines of the Gradle task '**test**' in the '**build.gradle**' corresponding to the JVM arguments and which point respectively to the library '**libglfw.dylib**' which fixes a MoltenVK bug in version 3.2.3 of LWJGL and the Vulkan library necessary when uses GLFW.
```
test {
	jvmArgs '-Dorg.lwjgl.glfw.libname=path-to libglfw.dylib'
	jvmArgs '-Dorg.lwjgl.vulkan.libname=path-to /vulkansdk-macos-1.1.126.0/macOS/lib/libvulkan.dylib'
	useJUnitPlatform()
}

```
[TestVulkan2.zip](https://github.com/LWJGL/lwjgl3/files/3974500/TestVulkan2.zip)

**Comment by Spasi** [comment]
Hey @Graou74,

>  which fixes a MoltenVK bug in version 3.2.3 of LWJGL

What bug is that? Also, the latest 3.2.4 snapshot includes an updated MoltenVK, could you please try it to see if the bug has been fixed?

**Comment by Graou74** [comment]
Hey @Spasi,

> What bug is that?

This one : [https://github.com/LWJGL/lwjgl3/issues/502]
Which is fixed in version LWJGL 3.2.4-SNAPSHOT

> Also, the latest 3.2.4 snapshot includes an updated MoltenVK, could you please try it to see if the bug has been fixed?

If you talk about the bug reported by this post, no it is not fixed. I have exactly the same behavior with LWJGL 3.2.4-SNAPSHOT as with versions 2.3.2 or 2.3.3.

**Comment by Spasi** [comment]
Hey @Graou74,

I'm not able to reproduce what you describe. Steps I followed:

1. Extracted your project, set LWJGL to `3.2.4-SNAPSHOT` and removed the jvm arguments.
2. Tested with `validation = false`, the application worked and the triangle was visible.
3. Downloaded the latest Vulkan SDK and did the following (as explained in the "Getting Started with the Vulkan SDK" documentation for macOS):
```
export VULKAN_SDK=<path to extracted SDK>/macos
export VK_ICD_FILENAMES=$VULKAN_SDK/etc/vulkan/icd.d/MoltenVK_icd.json
export VK_LAYER_PATH=$VULKAN_SDK/etc/vulkan/explicit_layer.d
export VK_LOADER_DEBUG=all
```
4. Added `jvmArgs '-Dorg.lwjgl.vulkan.libname=<path to extracted SDK>/macOS/lib/libvulkan.dylib'` to the gradle build. This is necessary so that layers can be detected and enabled by the Vulkan loader. By default LWJGL skips the loader and uses MoltenVK directly.
5. Tested with `validation = true`, the application worked without errors/warnings and the triangle was visible again.

The `VK_LOADER_DEBUG` environment variable isn't necessary, but useful to make sure everything works as expected.

I also tried LWJGL's `HelloVulkan` demo, debug output from the validation layers worked fine (that demo has known issues).
